skstore/ts/src/prv/skstore_skjson.ts

```diff
@@ -0,0 +1,429 @@
+    if (prop === "toJSON") return true;
+    const fields = hdl.objectFields();
+    return fields.has(prop);
+  },
+  ownKeys(hdl: WasmHandle) {
+    return Array.from(hdl.objectFields().keys());
+  },
+  getOwnPropertyDescriptor(hdl: WasmHandle, prop: string) {
+    const fields = hdl.objectFields();
+    const idx = fields.get(prop);
+    if (idx === undefined) return undefined;
+    const value = getValueAt(hdl, idx, false);
+    return {
+      configurable: true,
+      enumerable: true,
+      writable: false,
+      value,
+    };
+  },
+};
+
+export const reactiveArray = {
+  get(hdl: WasmHandle, prop: string, self: any): any {
+    if (typeof prop === "symbol") {
+      return () => self.toJSON().values();
+    } else {
+      if (prop === "__isArrayProxy") return true;
+      if (prop === "__pointer") return hdl.pointer;
+      if (prop === "length")
+        return hdl.access.SKIP_SKJSON_arraySize(hdl.pointer);
+      if (prop === "clone") return () => clone(self);
+      if (prop === "toJSON")
+        return () => {
+          const res: any[] = [];
+          const length: number = self.length;
+          for (let i = 0; i < length; i++) {
+            res.push(getValueAt(hdl, i, true));
+          }
+          return res;
+        };
+      if (prop === "forEach")
+        return <T>(
+          callbackfn: (value: T, index: number, array: T[]) => void,
+          thisArg?: any,
+        ) => {
+          self.toJSON().forEach(callbackfn, thisArg);
+        };
+      const v = parseInt(prop);
+      if (!isNaN(v)) return getValueAt(hdl, v, true);
+      return undefined;
+    }
+  },
+  has(hdl: WasmHandle, prop: string): boolean {
+    if (prop === "__isArrayProxy") return true;
+    if (prop === "__pointer") return true;
+    if (prop === "length") return true;
+    if (prop === "clone") return true;
+    if (prop === "toJSON") return true;
+    return false;
+  },
+  ownKeys(hdl: WasmHandle) {
+    const l = hdl.access.SKIP_SKJSON_arraySize(hdl.pointer);
+    const res = Array.from(Array(l).keys()).map((v) => "" + v);
+    return res;
+  },
+  getOwnPropertyDescriptor(hdl: WasmHandle, prop: string) {
+    const v = parseInt(prop);
+    if (isNaN(v)) return undefined;
+    const value = getValueAt(hdl, v, true);
+    return {
+      configurable: true,
+      enumerable: true,
+      writable: false,
+      value,
+    };
+  },
+};
+
+function clone<T>(value: T): T {
+  const aValue = value as any;
+  if (typeof value === "object") {
+    if (Array.isArray(value)) {
+      return value.map(clone) as T;
+    } else if (aValue.__isArrayProxy) {
+      const res: any[] = [];
+      const length: number = aValue.length;
+      for (let i = 0; i < length; i++) {
+        res.push(clone(aValue[i]));
+      }
+      return res as T;
+    } else if (aValue.__isObjectProxy) {
+      const res: any = {};
+      for (const key of aValue.keys) {
+        res[key] = clone(aValue[key]);
+      }
+      return res as T;
```

Comment:
Maybe I don't understand proxy objects well enough, but I expected that since there is the `ownKeys` function, the code below for non-proxy objects would work also for proxy objects. Am I confused?